### PR TITLE
Pytest raises

### PR DESCRIPTION
**Pull request by fabianegli**
- Use pytest.raises instead of custom try except blocks.
- Use touch instead of open and close to create empty files.
- Removed some useless else.

## PR checklist

- [x] This comment contains a description of changes (with reason)
- [ ] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated

### CODE REVIEW

**Review by mashehu — APPROVED** [review]
cc @Aratz for a more elegant way to create empty files.

**Comment by fabianegli** [comment]
@Aratz do you know why the `pytest tests/test_sync.py::TestModules::test_close_open_template_merge_prs` test fails locally? Or does it work in your local environment and it is just mine that is behaving strangely? (Link to relevant thread in [tools Slack channel](https://nfcore.slack.com/archives/CE5LG7WMB/p1665850129773939?thread_ts=1665850093.986309&cid=CE5LG7WMB))

File: tests/test_sync.py [hunk 1]
```diff
@@ -349,8 +350,7 @@ def test_close_open_template_merge_prs(self, mock_get):
             for pr in prs:
                 if pr["state"] != "open":
                     continue
```
Comment (Aratz) [inline tests/test_sync.py:352]:
```suggestion
                if pr["state"] == "open":
                    mock_close_open_pr.assert_any_call(pr)
```
This is probably the better way to write this

File: tests/test_sync.py [hunk 2]
```diff
@@ -349,8 +350,7 @@ def test_close_open_template_merge_prs(self, mock_get):
...
-                else:
-                    mock_close_open_pr.assert_any_call(pr)
+                mock_close_open_pr.assert_any_call(pr)
```
Comment (Aratz) [inline tests/test_sync.py:353]:
```suggestion
```

**Review by Aratz — APPROVED** [review]
LGTM :+1: 

Just had one suggestion where the code could actually be even more concise.

File: tests/test_sync.py (hunk 1 above)
Comment (fabianegli) [reply tests/test_sync.py]: yup. that's better.

**Pull request merged**